src/query.ts

```diff
@@ -140,6 +133,10 @@ async function run(): Promise<void> {
     // We can now delete the work dir. All required files have already been uploaded.
     chdir(curDir);
     fs.rmdirSync(workDir, { recursive: true });
+
+    // Save that we have already completed this repo so we don't set the state
+    // to failure in the post-action when a later repo fails.
+    saveState(`repo_${repo.id}_completed`, "true");
```

Comment:
I think we need to use `saveState` around https://github.com/github/codeql-variant-analysis-action/blob/56439308519c257fcebcafcb475b1dd34d1d7215/src/query.ts#L75 too. That code is handling if we hit an error downloading the query pack, so we mark the job as failed and update the state of each repo task. So if we do that there we don't want to do it again in the post action.

Reply:
Thanks! I've now added it and also moved this `saveState` call to be above the `chdir` call since we don't need to set the state to failed if that step fails (which is very unlikely anyway).